test-utils/UserIdTracker.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import scala.collection.mutable
+import domain.models.developers.UserId
+// import model.Collaborator
+// import model.CollaboratorRole._
+
+trait UserIdTracker {
+  private val idsByEmail = mutable.Map[String, UserId]()
+  
+  def idOf(email: String): UserId = idsByEmail.getOrElseUpdate(email, UserId.random)
+}
+
+// trait CollaboratorTracker extends UserIdTracker {
```

Comment:
Dead code from the copy OR Code we should be using to build collaborators

Reply:
Done